master/buildbot/worker/openstack.py

```diff
@@ -111,30 +112,81 @@ def _parseBlockDevice(self, block_device):
         client_block_device['uuid'] = block_device['uuid']
         client_block_device['boot_index'] = int(
             block_device.get('boot_index', 0))
-        client_block_device['volume_size'] = block_device['volume_size']
+        # Allow None here. It will be rendered later.
+        client_block_device['volume_size'] = block_device.get('volume_size')
         return client_block_device
 
-    @staticmethod
-    def _getImage(os_client, image):
+    @defer.inlineCallbacks
+    def _renderBlockDevice(self, block_device, build):
+        """Render all of the block device's values."""
+        rendered_block_device = yield build.render(block_device.copy())
```

Comment:
yes it does. You can convince yourself here:
https://github.com/buildbot/buildbot/blob/master/master/buildbot/process/properties.py#L796